test/junit/scala/tools/nsc/interpreter/CompletionTest.scala

```diff
@@ -206,6 +204,28 @@ class CompletionTest {
     )
   }
 
+  @Test
+  def isDeprecated(): Unit = {
+    val completer = setup()
+    // scala.Traversable is deprecated
+    val candidates1 = completer.complete("import scala.Traversab").candidates
+    assert(candidates1.size == 1)
+    assert(candidates1.forall(_.isDeprecated))
+    // StringOps#distinct and distinctBy are both deprecated
+    val candidates2 = completer.complete("\"foo\".distinc").candidates
+    assert(candidates2.size == 2)
+    assert(candidates2.forall(_.isDeprecated))
+  }
+
+  @Test
+  def isNotDeprecated(): Unit = {
+    val completer = setup()
+    // neither Growable nor GrowableBuilder are deprecated -- scala/bug#12043
+    val candidates = completer.complete("import scala.collection.mutable.Growable").candidates
+    assert(candidates.size == 2)
+    assert(candidates.forall(!_.isDeprecated))
+  }
+
```

Comment:
I'd prefer if you define test classes variations of `@deprecated` separate from the standard library.  I think they could just be defined in this file; if not they could be defined in a previous line of the REPL session.

Reply:
Indeed, in 5 years, 3.1 will remove `Traversable`, and in 10 years, 4.0 will rewrite collections entirely, no doubt deprecating `GrowableBuilder`!